easybuild/easyblocks/c/cudacompat.py

```diff
@@ -0,0 +1,263 @@
+                    raise EasyBuildError('Expected driver version %s (from nv_version) but found %s '
+                                         '(determined from file %s)', nv_version, library_version, library_path)
+
+            copy_file(library_path, os.path.join(libdir, library))
+            if library.endswith('.' + nv_version):
+                # E.g. libcuda.so.510.73.08 -> libcuda.so.1
+                versioned_symlink = library[:-len(nv_version)] + '1'
+            else:
+                # E.g. libnvidia-nvvm.so.4.0.0 -> libnvidia-nvvm.so.4
+                versioned_symlink = library.rsplit('.', 2)[0]
+            symlink(library, os.path.join(libdir, versioned_symlink), use_abspath_source=False)
+            # E.g. libcuda.so.1 -> libcuda.so
+            unversioned_symlink = versioned_symlink.rsplit('.', 1)[0]
+            symlink(versioned_symlink, os.path.join(libdir, unversioned_symlink), use_abspath_source=False)
+
+    def make_module_req_guess(self):
+        """Don't try to guess anything."""
+        return dict()
+
+    def make_module_extra(self):
+        """Skip the changes from the Binary EasyBlock and (only) set LD_LIBRARY_PATH."""
+
+        txt = super(Binary, self).make_module_extra()
+        txt += self.module_generator.prepend_paths('LD_LIBRARY_PATH', 'lib')
+        return txt
+
+    def sanity_check_step(self):
+        """Check for core files (unversioned libs, symlinks)"""
+        libraries = [
+            'libcuda.so',
+            'libnvidia-ptxjitcompiler.so',
+        ]
+        if LooseVersion(self.version) >= '11.5':
+            libraries.append('libnvidia-nvvm.so')
+        custom_paths = {
+            'files': [os.path.join(self.installdir, 'lib', x) for x in libraries],
+            'dirs': ['lib', 'lib64'],
+        }
+        super(EB_CUDAcompat, self).sanity_check_step(custom_paths=custom_paths)
+
+        if self.has_nvidia_smi:
+            fake_mod_data = None
+
+            # skip loading of fake module when using --sanity-check-only, load real module instead
+            if build_option('sanity_check_only'):
+                self.load_module()
+            elif not self.dry_run:
+                fake_mod_data = self.load_fake_module(purge=True, verbose=True)
+
+            try:
+                out_lines = self._run_nvidia_smi('--query --display=COMPUTE')
+
+                if fake_mod_data:
+                    self.clean_up_fake_module(fake_mod_data)
+
+                cuda_version = next((line.rsplit(' ', 1)[1] for line in out_lines if line.startswith('CUDA')), None)
+                if not cuda_version:
+                    raise RuntimeError('Failed to find CUDA version!')
+                self.log.info('CUDA version (as per nvidia-smi) after loading the module: ' + cuda_version)
+                if LooseVersion(cuda_version) != self.version:
+                    raise RuntimeError('Reported CUDA version %s is not %s' % (cuda_version, self.version))
```

Comment:
@Flamefire This check will always fail is someone is filtering out `LD_LIBRARY_PATH` from the final module file (as is common if using `--rpath`, see https://docs.easybuild.io/en/latest/RPATH-support.html#relation-to-ld-library-path). We should check (early on?) that this setting is not being used as in this case you really must use `LD_LIBRARY_PATH` for this to work as intended.
```
if 'LD_LIBRARY_PATH' in (build_option('filter_env_vars') or []):
    ...
```

Reply:
Added that check